### ISSUE TITLE
Consider computing `toString` values for `dart.str` inline

### ISSUE THREAD

**Issue by leafpetersen** [dart-lang/sdk#30511]
Not 100% sure this is a valid optimization, but we can often generate a direct call to `.toString` if we know the receiver type, which we never will within `dart.str`.  So calling the `.toString` method on the arguments to `dart.str` instead of within the method might be more efficient.

**Comment by jmesserly** [comment]
agreed. I think template strings might be kinda slow and not worth it in some cases.

FWIW, I did make `dart.toString` faster recently.
